pandas/tests/series/indexing/test_setitem.py

```diff
@@ -1665,3 +1665,14 @@ def test_setitem_empty_mask_dont_upcast_dt64():
     ser.mask(mask, "foo", inplace=True)
     assert ser.dtype == dti.dtype  # no-op -> dont upcast
     tm.assert_series_equal(ser, orig)
+
+
+def test_setitem_multi_dimension_array_to_dataframe():
+    # GH#40827
+    df = DataFrame(np.zeros((4, 1)), columns=["A"])
+    y = np.random.randn(4, 3)
```

Comment:
What happens here, if the column A has duplicates?

Reply:
Sorry I do not fully understand this. 
`column A has duplicates`
Are you mean that column A has duplicated values like this?

df
#           A
# 0  1.2
# 1 1.2
# 2 1.2
# 3  1.2

Reply:
Sorry bad wording, I meant a df with duplicated column names

Reply:
If df with duplicated column names, with version 1.5.0.dev0+1217.g6e1a040142
```
df = pd.DataFrame(numpy.zeros((4, 2)), columns=['A', 'A'])
y = numpy.random.randn(4, 3)

df['A'] = y

print(df)
```
Result
```
          A         A
0 -0.256262  0.197093
1 -0.628350 -0.111466
2  0.394070  0.111357
3  0.780353 -0.20774
```

Reply:
The interesting part is when the rhs has as many columns as A is duplicated

Reply:
When the rhs has as many columns as A is duplicated,
```
df = DataFrame(
    [[1, 2, 3, 4], [1, 2, 3, 4]],
    columns=["A", "A", "A", "A"],
)
df['A'] = np.array([[5, 6, 7, 8], [5, 6, 7, 8]])
print(df)
```
Result:
```
   A  A  A  A
0  5  6  7  8
1  5  6  7  8

```